src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,126 @@
+                                Task task = new Deadline(workToDo, LocalDateTime.parse(deadline, formatter));
+                                tasks.addTask(task);
+                                ui.printAddTask(task, tasks);
+                            }
+                        } else if (str.startsWith("event")) {
+                            if (!str.contains("from")) {
+                                throw new EventCommandUseException(str);
+                            } else {
+                                String fromMarker = "/from "; //mark the /from index of the string
+                                int firstIndex = str.indexOf(fromMarker);
+                                int secondIndex;
+                                String fromWhen;
+                                String toWhen;
+                                String workToDo = str.substring(6, firstIndex);
+                                String afterFirstIndex = str.substring(firstIndex + 6);
+                                if (!afterFirstIndex.contains("/to ")) { //to check the input of /to after /from
+                                    throw new EventCommandUseException(str);
+                                } else {
+                                    String toMarker = "/to "; //mark the /to index of the string
+                                    secondIndex = afterFirstIndex.indexOf(toMarker); //to make sure we get the /to after the /from
+                                    fromWhen = afterFirstIndex.substring(0, secondIndex).trim(); //get the from timing
+                                    toWhen = afterFirstIndex.substring(secondIndex + 3).trim(); //get the to timing
+                                    if (fromWhen.trim().isEmpty() ||
+                                            toWhen.trim().isEmpty()) { //needs to check whether there is anything after /by
+                                        throw new EventCommandUseException(str);
+                                    }
+                                    Task task = new Event(workToDo, LocalDateTime.parse(fromWhen, formatter),
+                                            LocalDateTime.parse(toWhen, formatter));
+                                    tasks.addTask(task);
+                                    ui.printAddTask(task, tasks);
+                                }
+                            }
+                        } else {
+                            throw new InvalidInputException(str);
+                        }
+                    }
+                } else {
+                    ui.listTasks(tasks);
+                }
+        } catch (java.time.format.DateTimeParseException e) {
+            //detect inputs that don't follow the yyyy-MM-dd HHmm format
+            ui.printException();
+        } catch (InvalidInputException | EventCommandUseException |
+                 DeadlineCommandUseException | ToDoCommandUseException e) {
+            ui.printException(e.getMessage());
```

Comment:
Perhaps you should break the line before the operator symbol, e.g.
```suggestion
        } catch (InvalidInputException | EventCommandUseException
                | DeadlineCommandUseException | ToDoCommandUseException e) {
            ui.printException(e.getMessage());
```